src/lib/swapper/swappers/CowSwapper/cowExecuteTrade/cowExecuteTrade.test.ts

```diff
@@ -57,7 +42,34 @@ jest.mock('../utils/helpers/helpers', () => {
   }
 })
 
-const ethereumMock = jest.mocked(ethereum)
+const actualChainAdapters = jest.requireActual('@shapeshiftoss/chain-adapters')
+
+const mockEthereum = {
+  getBIP44Params: jest.fn(() => actualChainAdapters.ethereum.ChainAdapter.defaultBIP44Params),
+  getChainId: () => KnownChainIds.EthereumMainnet,
+  signMessage: jest.fn(() => Promise.resolve(Signature)),
+} as unknown as ethereum.ChainAdapter
+
+const mockGnosis = {
+  getBIP44Params: jest.fn(() => actualChainAdapters.gnosis.ChainAdapter.defaultBIP44Params),
+  getChainId: () => KnownChainIds.GnosisMainnet,
+  signMessage: jest.fn(() => Promise.resolve(Signature)),
+} as unknown as ethereum.ChainAdapter // wrong type but doesn't matter here
+
+jest.mock('context/PluginProvider/chainAdapterSingleton', () => {
+  const { KnownChainIds } = require('@shapeshiftoss/types')
```

Comment:
Can we avoid requiring actual by using `CowChainId` ?

Reply:
Not really, it's a type so we still need to require it. We could use just a raw string but I'd rather keep it type safe